doc/whats_new/v1.3.rst

```diff
@@ -128,6 +128,13 @@ Changelog
   when the provided `sample_weight` reduces the problem to a single class in `fit`.
   :pr:`24140` by :user:`Jonathan Ohayon <Johayon>` and :user:`Chiara Marmo <cmarmo>`.
 
+:mod:`sklearn.neighbors`
+........................
+
+- |Fix| :class:`neighbors.NearestNeighbors` and other similar neighbors searches
```

Comment:
I would consider this change as an "Enhancement". We were not officially supporting `"nan_euclidean"`.
I think that we can also put the list of all estimators to be exhaustive. It is always more informative for our end-user.